cms/topics/blocks.py

```diff
@@ -164,3 +178,69 @@ def media(self) -> forms.Media:
 
 
 register(SeriesWithHeadlineChooserAdapter(), TopicHeadlineFigureBlock)
+
+
+class TimeSeriesPageLinkBlock(StructBlock):
+    title = CharBlock(required=True)
+    description = TextBlock(required=True)
+    url = URLBlock(
+        required=True,
+        help_text="The URL must start with 'https://' "
+        f"and match one of the allowed domains or their subdomains: {', '.join(settings.ONS_ALLOWED_LINK_DOMAINS)}",
+    )
+
+    class Meta:
+        icon = "link"
+        template = "templates/components/streamfield/time_series_link.html"
+
+    def clean(self, value: "StructValue") -> "StructValue":
+        """Checks that the given time series page URL matches the allowed domain."""
+        errors = {}
+        parsed_url = urlparse(value["url"])
+
+        if not parsed_url.hostname or parsed_url.scheme != "https":
+            errors["url"] = ValidationError(
+                "Please enter a valid URL. It should start with 'https://' and contain a valid domain name."
+            )
+        elif not any(
+            is_hostname_in_domain(parsed_url.hostname, allowed_domain)
+            for allowed_domain in settings.ONS_ALLOWED_LINK_DOMAINS
+        ):
+            patterns_str = " or ".join(settings.ONS_ALLOWED_LINK_DOMAINS)
+            errors["url"] = ValidationError(
+                f"The URL hostname is not in the list of allowed domains or their subdomains: {patterns_str}"
+            )
+
+        if errors:
+            raise StructBlockValidationError(block_errors=errors)
+
+        return super().clean(value)
+
+
+class TimeSeriesPageStoryBlock(StreamBlock):
+    time_series_page_link = TimeSeriesPageLinkBlock()
+
+    def clean(self, value: "StreamValue", ignore_required_constraints: bool = False) -> "StreamValue":
+        cleaned_value = super().clean(value)
+
+        # For each time series URL, record the indices of the blocks it appears in
+        urls = defaultdict(set)
+        for block_index, block in enumerate(cleaned_value):
+            url = block.value["url"].lower().rstrip("/")  # Treat URLs with and without trailing slashes as equivalent
+
+            url = url.removeprefix("https://").removeprefix("www.")  # Normalize the URL
```

Comment:
Nit: You don't need it separately as we enforce `https`.
```suggestion
            url = url.removeprefix("https://www.")  # Normalize the URL
```

Reply:
Not you're not, I am. Good shout. Okay, LGTM 👍